Doc/library/compression.zstd.rst

```diff
@@ -0,0 +1,834 @@
+
+   .. attribute:: hash_log
+
+      Size of the initial probe table, as a power of two. The resulting memory
+      usage is ``1 << (hash_log+2)`` bytes. Larger tables improve compression
+      ratio of strategies <= :attr:`~Strategy.dfast`, and improve compression
+      speed of strategies > :attr:`~Strategy.dfast`.
+
+      A value of zero causes the value to be selected automatically.
+
+   .. attribute:: chain_log
+
+      Size of the multi-probe search table, as a power of two. The resulting
+      memory usage is ``1 << (chain_log+2)`` bytes. Larger tables result in
+      better and slower compression. This parameter has no effect for the
+      :attr:`~Strategy.fast` strategy. It's still useful when using
+      :attr:`~Strategy.dfast` strategy, in which case it defines a secondary
+      probe table.
+
+      A value of zero causes the value to be selected automatically.
+
+   .. attribute:: search_log
+
+      Number of search attempts, as a power of two. More attempts result in
+      better and slower compression. This parameter is useless for
+      :attr:`~Strategy.fast` and :attr:`~Strategy.dfast` strategies.
+
+      A value of zero causes the value to be selected automatically.
+
+   .. attribute:: min_match
+
+      Minimum size of searched matches. Larger values increase compression and
+      decompression speed, but decrease ratio. Note that Zstandard can still
+      find matches of smaller size, it just tweaks its search algorithm to look
+      for this size and larger. For all strategies < :attr:`~Strategy.btopt`,
+      the effective minimum is ``4``; for all strategies
+      > :attr:`~Strategy.fast`, the effective maximum is ``6``.
+
+      A value of zero causes the value to be selected automatically.
+
+   .. attribute:: target_length
+
+      The impact of this field depends on the selected :class:`Strategy`.
+
+      For strategies :attr:`~Strategy.btopt`, :attr:`~Strategy.btultra` and
+      :attr:`~Strategy.btultra2`, the value is the length of a match
+      considered "good enough" to stop searching. Larger values make
+      compression ratios better, but compresses slower.
+
+      For strategy :attr:`~Strategy.fast`, it is the distance between match
+      sampling. Larger values make compression faster, but with a worse
+      compression ratio.
+
+      A value of zero causes the value to be selected automatically.
+
+   .. attribute:: strategy
+
+      The higher the value of selected strategy, the more complex the
+      compression technique used by zstd, resulting in higher compression
+      ratios but slower compression.
+
+      .. seealso:: :class:`Strategy`
+
+   .. attribute:: enable_long_distance_matching
+
+      Long distance matching can be used to improve compression for large
+      inputs by finding large matches at greater distances. It increases memory
+      usage and window size.
+
+      ``True`` or ``0`` enable long distance matching while ``False`` or ``1``
```

Comment:
This is VERY counterintuitive and error prone.

Reply:
I fixed the 3 instances of this.